Task5Test/src/test/java/com/company/TriangleTest.java

```diff
@@ -3,75 +3,94 @@
-                {BigDecimal.valueOf(11), BigDecimal.valueOf(10), BigDecimal.valueOf(10)},
-        };
-    }
-
-    @Test(dataProvider = "ValidIsoscelesTriangle")
-    public void testPositiveSetValidSidesOfTheIsoscelesTriangle(BigDecimal a, BigDecimal b, BigDecimal c) throws Exception {
-        triangle = new Triangle(a, b, c);
-        Assert.assertEquals(triangle.definesType(), Triangle.ISOSCELES);
-    }
-
-
-    @DataProvider(name = "setInvalidSides")
-    public Object[][] setNotValidSidesOfTheTriangle() {
-        return new Object[][] {
-                {BigDecimal.valueOf(0), BigDecimal.valueOf(1), BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(0), BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), BigDecimal.valueOf(0)},
+    private static final String VALID_TRIANGLES = "validTriangles";
+    private static final String INVALID_TRIANGLES = "invalidTriangles";
+    private static final String ISOSCELES_NODE = "isosceles";
+    private static final String EQUILATERAL_NODE = "equilateral";
+    private static final String SIDE_A = "a";
+    private static final String SIDE_B = "b";
+    private static final String SIDE_C = "c";
 
-                {BigDecimal.valueOf(-1), BigDecimal.valueOf(1), BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(-1), BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), BigDecimal.valueOf(-1)},
 
-                {null, BigDecimal.valueOf(1), BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), null, BigDecimal.valueOf(1)},
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), null},
-        };
-    }
-
-    @Test(dataProvider = "setInvalidSides", expectedExceptions = Exception.class)
-    public void testNegativeSetNotValidSidesOfTheTriangle(BigDecimal a, BigDecimal b, BigDecimal c) throws Exception {
-        triangle = new Triangle(a, b, c);
-    }
+    Triangle triangle;
 
-    @DataProvider(name = "ValidEquilateralTriangle")
-    public Object[][] setValidSidesOfTheEquilateralTriangle() {
-        return new Object[][]{
-                {BigDecimal.valueOf(1), BigDecimal.valueOf(1), BigDecimal.valueOf(1)}
-        };
+    @DataProvider(name = "valid triangles from xml")
+    public Object[][] readXMLforPositiveIsosceles() throws Exception {
+        File inputFile = new File("/triangleData.xml");
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+
+        Document document = builder.parse(inputFile);
+        Node validTriangleElement = document.getElementsByTagName(VALID_TRIANGLES).item(0);
+        NodeList nodes = validTriangleElement.getChildNodes();
+        for (int i = 0; i < nodes.getLength(); i++) {
+            if (Node.TEXT_NODE == nodes.item(i).getNodeType()) {

```

Comment:
It looks strange, I doubt if you need such magic just to get some values from xml